src/file.cr

```diff
@@ -119,6 +119,23 @@ class File < IO::FileDescriptor
     accessible?(path, LibC::F_OK)
   end
 
+  # Returns `true` if the file at *path* is empty, otherwise returns `false`.
+  # Raises `Errno` if the file at *path* does not exist.
+  #
+  # ```
+  # File.write("foo", "")
+  # File.empty?("foo") # => true
+  # File.write("foo", "foo")
+  # File.empty?("foo") # => false
+  # ```
+  def self.empty?(path) : Bool
+    begin
+      stat(path).size == 0
+    rescue Errno
+      raise Errno.new("Error determining size of '#{path}'")
```

Comment:
This captures and re-raises the error, with a different message. I don't know if there's a more elegant / obvious way to do this.